xen/drivers/passthrough/coproc.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * Code to interact with a coproc device described as a device tree node
+ *
+ * Oleksandr Tyshchenko <Oleksandr_Tyshchenko@epam.com>
+ * Copyright (C) 2017 EPAM Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include <xen/lib.h>
+#include <xen/sched.h>
+#include <xen/guest_access.h>
+#include <xen/iommu.h>
+#include <xen/device_tree.h>
```

Comment:
hmm, it is going to be difficult at the moment, because the scf as well as IOMMU on ARM rely on device-tree. But, need to think about it.